moo.js

```diff
@@ -571,13 +573,26 @@
         col: this.col,
       }
     }
-    var start = Math.max(0, token.offset - token.col + 1)
-    var eol = token.lineBreaks ? token.text.indexOf('\n') : token.text.length
-    var firstLine = this.buffer.substring(start, token.offset + eol)
+    
+    var lines = this.buffer.split("\n")
```

Comment:
We avoid using `split` here because we've already tokenized the input. Error handling doesn't necessarily need to be as fast as the happy path, but it could potentially be an issue if the input buffer is large.

Reply:
> We avoid using `split` here because we've already tokenized the input. Error handling doesn't necessarily need to be as fast as the happy path, but it could potentially be an issue if the input buffer is large.

We could walk the string backwards instead if that is a concern.

Reply:
Walking backwards (and forwards—see my comment) is a better option. Tokenizers often deal with very large inputs.